src/main/scala/carldata/series/Stats.scala

```diff
@@ -29,11 +29,33 @@ object Stats {
     else vs.sum / vs.length
   }
 
-  /** Calculate covariance between 2 places in the series */
+  /** Calculate correlation between 2 places in the series */
   def autoCorrelation[V: Fractional](values: Seq[V], pos1: Int, pos2: Int, size: Int): Double = {
     val MeanAndVariance(mean, variance) = meanAndVariance(values)
     if(variance == 0) 0
     else autoCovariance(values, pos1, pos2, size, mean) / variance
   }
 
+  /** Calculate covariance with two-pass algorithm. */
+  def covariance[V: Fractional](xs: Seq[V], ys: Seq[V])(implicit num: Fractional[V]): V = {
+    def sumFractional(vs: Seq[V]): V = {
```

Comment:
Check how the mean is calculated in meanAndVariance function.
There is no need for this function.
And even if it was needed that it should be made tail recursive.